presto-main/src/main/java/com/facebook/presto/execution/scheduler/LegacySqlQueryScheduler.java

```diff
@@ -459,9 +505,182 @@ else if (!result.getBlocked().isDone()) {
                 // get all sections ready for execution
                 .filter(this::isReadyForExecution)
                 .limit(maxConcurrentMaterializations - runningPlanSections)
+                .map(this::tryCostBasedOptimize)
                 .collect(toImmutableList());
     }
 
+    /**
+     * Utility function to invoke runtime cost-based optimizer, which determines if the probe and build side of a JoinNode should be swapped,
```

Comment:
nit: this is a general purpose utility function (it just happens to be that right now there is only one plan optimizer)